src/main/java/emissary/core/channels/LoggingChannelFactory.java

```diff
@@ -0,0 +1,205 @@
+         */
+        private final boolean logStackTrace;
+        /**
+         * A one-up counter for identifying each instance.
+         */
+        private final AtomicLong currentInstance = new AtomicLong(0);
+
+        private LoggingChannelFactoryImpl(final SeekableByteChannelFactory seekableByteChannelFactory,
+                final String identifier, final Logger logger, final boolean logStackTrace) {
+            Validate.notNull(seekableByteChannelFactory, "Required: seekableByteChannelFactory not null!");
+            Validate.notNull(identifier, "Required: identifier not null!");
+            Validate.notNull(logger, "Required: logger not null!");
+
+            this.seekableByteChannelFactory = seekableByteChannelFactory;
+            this.identifier = identifier;
+            this.logger = logger;
+            this.logStackTrace = logStackTrace;
+        }
+
+        @Override
+        public SeekableByteChannel create() {
+            return new LoggingSeekableByteChannel(seekableByteChannelFactory.create(),
+                    identifier + " : " + currentInstance.getAndIncrement(), logger, logStackTrace);
+        }
+    }
+
+    /**
+     * This SeekableByteChannel wraps another SeekableByteChannel and logs all methods calls.
+     */
+    private static class LoggingSeekableByteChannel implements SeekableByteChannel {
+        /**
+         * The SeekableByteChannel that is being wrapped.
+         */
+        private final SeekableByteChannel seekableByteChannel;
+        /**
+         * The prefix added to the log statement.
+         */
+        private final String prefix;
+        /**
+         * The logger to be used for logging.
+         */
+        private final Logger logger;
+        /**
+         * Determines whether a stack trace should also be logged.
+         */
+        private final boolean logStackTrace;
+
+        /**
+         * Constructs a SeekableByteChannel that wraps another SeekableByteChannel and logs all methods calls.
+         * 
+         * @param seekableByteChannel to be wrapped.
+         * @param prefix added to the log statement.
+         * @param logger to be used for logging.
+         * @param logStackTrace determines whether a stack trace should also be logged.
+         */
+        private LoggingSeekableByteChannel(final SeekableByteChannel seekableByteChannel, final String prefix,
+                final Logger logger, final boolean logStackTrace) {
+            this.seekableByteChannel = seekableByteChannel;
+            this.prefix = prefix;
+            this.logger = logger;
+            this.logStackTrace = logStackTrace;
+
+            log(logger, logStackTrace, "{} : created : lST={}", prefix, logStackTrace);
+        }
+
+        @Override
+        public boolean isOpen() {
+            final boolean open = seekableByteChannel.isOpen();
+
+            log(logger, logStackTrace, "{} : isOpen : r={}", prefix, open);
```

Comment:
```suggestion
            log(logger, logStackTrace, "{} : isOpen : open={}", prefix, open);
```

Reply:
Changed to 'o' to be similar to the other comments.